instrumentation/src/main/java/io/opentelemetry/android/emitter/EventEmitterWithData.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.emitter;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.events.EventEmitter;
+import java.util.Map;
+import java.util.Objects;
+
+class EventEmitterWithData {
+
+    private static final String EVENT_DATA = "event.data";
+
+    private final EventEmitter delegate;
+
+    EventEmitterWithData(EventEmitter delegate) {
+        this.delegate = delegate;
+    }
+
+    public void emit(String eventName, Attributes attributes, Map<String, Object> data) {
+        Attributes dataAsAttribute = convertRichDataObjectIntoAttributes(data);
+        Attributes eventAttrs =
+                attributes.toBuilder().put(EVENT_DATA, dataAsAttribute.toString()).build();
+        delegate.emit(eventName, eventAttrs);
+    }
+
+    private Attributes convertRichDataObjectIntoAttributes(Map<String, Object> data) {
```

Comment:
Let's double check this - Based on the data we expect in event.data for Android, are we sure we're okay to have these 4 types only - String, Long, Double, Boolean and all other types could be correctly represented by calling toString() [line 48] (As in it will not end up printing the memory address but will give actual content (value) of the attribute for any other types we expect)

Reply:
We could add other types yes, but at least when considering what this is currently planned to be used for (emitting crashes), I don't anticipate other data types unless I missed one and you can think of any? And if there are other object types, its possible that they'll have their own toString() method which will give an appropriate string representation of the object. 

And if not, it will resort to the default behaviour of toString() is to print class name, then @, then unsigned hexadecimal representation of the hash code of the object. Which I think is acceptable in an edge case. As if its not an object type we expect, whatever backend may not be able to parse it anyway. I'm definitely all ears and open to an alternative approach though if you have any ideas!